src/lib/operators/table_scan/between_table_scan_impl.cpp

```diff
@@ -16,81 +16,68 @@
 
 namespace opossum {
 
-BetweenTableScanImpl::BetweenTableScanImpl(const std::shared_ptr<const Table>& in_table, const ColumnID left_column_id,
+BetweenTableScanImpl::BetweenTableScanImpl(const std::shared_ptr<const Table>& in_table, const ColumnID column_id,
                                            const AllTypeVariant& left_value, const AllTypeVariant& right_value)
-    : BaseSingleColumnTableScanImpl{in_table, left_column_id, PredicateCondition::Between},
+    : AbstractSingleColumnTableScanImpl{in_table, column_id, PredicateCondition::Between},
       _left_value{left_value},
       _right_value{right_value} {}
 
 std::string BetweenTableScanImpl::description() const { return "BetweenScan"; }
 
-void BetweenTableScanImpl::handle_segment(const BaseValueSegment& base_segment,
-                                          std::shared_ptr<SegmentVisitorContext> base_context) {
-  auto context = std::static_pointer_cast<Context>(base_context);
-  auto& matches_out = context->_matches_out;
-  const auto& position_filter = context->_position_filter;
-  const auto chunk_id = context->_chunk_id;
-
-  // TODO(anyone): A lot of code is duplicated here, below, and in the other table scans.
-  // This can be improved once we have #1145.
-
-  const auto left_column_type = _in_table->column_data_type(_left_column_id);
-
-  resolve_data_type(left_column_type, [&](auto type) {
-    using ColumnDataType = typename decltype(type)::type;
-
-    auto& left_segment = static_cast<const ValueSegment<ColumnDataType>&>(base_segment);
-
-    auto left_segment_iterable = create_iterable_from_segment(left_segment);
+void BetweenTableScanImpl::_on_scan(const BaseSegment& segment, const ChunkID chunk_id, PosList& matches,
+                                    const std::shared_ptr<const PosList>& position_filter) const {
+  // early outs for specific NULL semantics
+  if (variant_is_null(_left_value) || variant_is_null(_right_value)) {
+    /**
+     * Comparing anything with NULL (without using IS [NOT] NULL) will result in NULL.
+     * Therefore, these scans will always return an empty position list.
+     * Because OpIsNull/OpIsNotNull are handled separately in IsNullTableScanImpl,
```

Comment:
Imo the bottom two lines of this comment aren't necessary. We are not in anything related to IS NULL here, why imply, and this is what happens when this is mentioned, that this class/function has anything to do with IS NULL? We are in a BETWEEN scan, and I think the reader can perfectly well imagine `x BETWEEN NULL AND y` in his head. 

The same goes for the other place where this comment is repeated.